src/components/timeline.js

```diff
@@ -0,0 +1,385 @@
+        fetchMergesByProcessAndDate(props.name, minDate, maxDate).then(
+            (merges) => {
+                if (merges.length > 0) {
+                    setShowMessage(false);
+                    dispatch(currentMergesList(merges));
+                    mergesForTimeline(merges);
+
+                    const lastMerge = merges[merges.length - 1];
+
+                    dispatch(
+                        updateMergeDate(props.name, new Date(lastMerge.date))
+                    );
+                    lastMerge.igms.forEach((igm) => {
+                        const status = lastMerge.status
+                            ? lastMerge.status
+                            : igm.status;
+                        updateIgm(igm.tso, toIgmStatus(status));
+                    });
+                } else {
+                    setShowMessage(true);
+                    dispatch(updateMergeDate(props.name, null));
+                }
+            }
+        );
+    };
+
+    /**
+     * Send the new selected date in datepicker to fetch merges
+     * @param e
+     */
+    const fetchMergesOnChangeDate = (e) => {
+        const minDate = e.target.value + minHour;
+        const maxDate = e.target.value + maxHour;
+        dispatch(currentMergesList(''));
+        getMergesByProcessAndDate(minDate, maxDate);
+    };
+
+    /**
+     * Send the current date to fetch merges
+     */
+    const fetchMergesByCurrentDay = () => {
+        const minDate = currentDateFormat() + minHour;
+        const maxDate = currentDateFormat() + maxHour;
+        getMergesByProcessAndDate(minDate, maxDate);
+    };
+
+    /**
+     * Handler click when change slider position
+     */
+    const handleChangeSlider = (e, val) => {
+        merges.forEach((merge) => {
+            const convertHour = convertHoursToMinutes(merge);
+            if (convertHour === val) {
+                dispatch(updateMergeDate(props.name, new Date(merge.date)));
+                if (merge.igms) {
+                    merge.igms.forEach((igm) => {
+                        const status = merge.status ? merge.status : igm.status;
+                        updateIgm(igm.tso, toIgmStatus(status));
+                    });
+                }
+            }
+        });
+    };
+
+    /**
+     * Convert total minutes to hours and minutes
+     * @param value
+     * @returns {string}
+     */
+    const valueLabelFormat = (value) => {
+        const hours = Math.floor(value / 60);
+        const minutes = value % 60;
+        return hours + 'h' + minutes;
+    };
+
+    /**
+     * Convert simple hour to total minutes
+     * @param merge
+     * @returns {number}
+     */
+    const convertHoursToMinutes = (merge) => {
+        const hour = merge.date.split('T')[1].split(':')[0];
+        const min = merge.date.split('T')[1].split(':')[1];
+        const convertHour = hour * 60 + parseInt(min);
+        return convertHour;
+    };
+
+    function connectNotifications(props) {
```

Comment:
So now both process and timeline components are connected to web socket ?
Why?

Reply:
Timeline connected to the websocket to update it that there is a new merge

Reply:
But if process is already connected timeline don t need. Process has to update timeline using props